.github/workflows/ci.yml

```diff
@@ -19,9 +19,42 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        Dtest: [ "A*,B*", "C*", "nonexistingtest" ]
+        Dtest: [ "FailA*", "FailB*", "PassA*" ]
     uses: ./.github/workflows/worker.yml
     with:
       script: ".github/scripts/run-maven-command.sh"
       args: "test -Dtest='${{ matrix.Dtest }}' -Dsurefire.failIfNoSpecifiedTests=false -P skip-static-checks -Dweb.console.skip=true -Dmaven.javadoc.skip=true --fail-at-end"
       jdk: ${{ inputs.jdk }}
+      upload_surefire_reports: true
+
+  reporting:
+    name: "report-unit-test-failures"
+    needs: run-unit-tests
+    runs-on: ubuntu-latest
+    if: ${{ always() }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Download individual surefire reports
+        uses: actions/download-artifact@v4
+        with:
+          pattern: "surefire-reports-*"
+          path: target/surefire-reports
+
+      - name: Flatten surefire reports
+        run: |
+          # Copy all files from subdirectories to the target directory
+          find target/surefire-reports -mindepth 2 -type f -exec mv {} target/surefire-reports/ \;
+          
+          # Remove the empty subdirectories
+          find target/surefire-reports -type d -empty -delete
+
+      - name: Print files in surefire-reports
+        run: |
+          ls -l target/surefire-reports
+          cat target/surefire-reports/*.xml
+
+      - name: Publish Test Report
+        if: ${{ always() }}
+        uses: scacap/action-surefire-report@v1
```

Comment:
looks good - not sure how this reporting tool will scale; for a few errors it looks promising.
we will need something which could handle say a 1000 failure without being broken

Reply:
Yeah, would like to do a larger scale testing. Let me know if the other changes look good, will then trigger some larger set of failures.

Reply:
it could work - make sure to break at least a 1000 testcases  :D